eosclubhouse/system.py

```diff
@@ -679,16 +680,61 @@ def is_running(self):
         '''
         return self.get_gtk_app_proxy().props.g_name_owner is not None
 
-    def is_installed(self):
-        '''Check if the app is installed.
+    def _is_flatpak_installed(self):
+        '''Check if the app is installed as flatpak.
 
         Note: This only works for apps distributed as flatpaks.
         '''
+
         result = subprocess.run(['/usr/bin/flatpak-spawn', '--host',
                                  'flatpak', 'info', '--show-ref', self.dbus_name],
                                 capture_output=True)
         return result.returncode == 0
 
+    def _get_app_info_path(self):
+        '''Look for the app desktop file on the system
+        '''
+
+        # The data_dirs array is based on the glib code to look for desktop files:
+        # https://gitlab.gnome.org/GNOME/glib/-/blob/master/gio/gdesktopappinfo.c#L1513
+
+        # We can't use `g_app_info_get_all` because it doesn't give us the apps
+        # that are outside of the flatpak sandbox.
+
+        data_dirs = [
+            # First, the configs. Highest priority: the user's ~/.config
+            GLib.get_user_config_dir(),
+            # Next, the system configs (/etc/xdg, and so on).
+        ] + GLib.get_system_config_dirs() + [
+            # Now the data. Highest priority: the user's ~/.local/share/applications
+            GLib.get_user_data_dir(),
+            # Following that, XDG_DATA_DIRS/applications, in order
+        ] + GLib.get_system_data_dirs() + [
```

Comment:
I checked all the returned dirs, and the only desktop files found were `com.hack_computer.Clubhouse.desktop` and ` com.hack_computer.HackToolbox.desktop`. So I don't think this works inside the sandbox.